pkg/lifecycle/service.go

```diff
@@ -176,6 +211,44 @@ func (s *Service) Start(
 	return nil
 }
 
+// StartWithBackoff starts a pipeline with a backoff.
+// It'll check the number of times the pipeline has been restarted and the duration of the backoff.
+// When the pipeline has reached out the maximum number of retries, it'll return a fatal error.
+func (s *Service) StartWithBackoff(ctx context.Context, rp *runnablePipeline) error {
```

Comment:
@raulb WDYT about just calling it `Restart`?

Reply:
@hariso That could work

Reply:
@hariso actually, ..., in my opinion, `Restart` would be `Stop` + `Start`. This method doesn't stop, but mainly start with an additional backoff so I think this method's name works.

Reply:
Ah, you're right, StartWithBackoff definitely makes more sense now that you said that.:)